src/header/components/HeaderSearch.tsx

```diff
@@ -86,34 +86,37 @@ const HeaderSearch: FunctionComponent = () => {
     }
   }
 
-  const onFormSubmit = (e: React.FormEvent<HTMLFormElement>) => {
-    e.preventDefault()
-    e.stopPropagation()
-
-    // If a suggestion is selected use that one, otherwise submit the form
-    if (selectedElement) {
-      document.querySelector<HTMLAnchorElement>(`.${ACTIVE_ITEM_CLASS}`)?.click()
-    } else {
-      const queryString = {
-        term: inputValue.trim(),
+  const onFormSubmit = useCallback(
+    (e: React.FormEvent<HTMLFormElement>) => {
+      e.preventDefault()
+      e.stopPropagation()
+
+      // If a suggestion is selected use that one, otherwise submit the form
+      if (selectedElement) {
+        document.querySelector<HTMLAnchorElement>(`.${ACTIVE_ITEM_CLASS}`)?.click()
+      } else {
+        const queryString = {
+          term: inputValue.trim(),
+        }
+        const query = new URLSearchParams(
+          Object.entries(queryString).reduce(
+            (acc, [key, value]) => (value ? { ...acc, [key]: value } : acc),
+            {},
+          ),
+        )
+        const actionType = Object.values(SEARCH_PAGE_CONFIG).find(
+          ({ type: configType }) => searchBarFilterValue === configType,
+        )
+        if (actionType) {
+          history.push({ pathname: actionType.path, search: query ? `?${query}` : '' })
+        }
       }
-      const query = new URLSearchParams(
-        Object.entries(queryString).reduce(
-          (acc, [key, value]) => (value ? { ...acc, [key]: value } : acc),
-          {},
-        ),
-      )
-      const actionType = Object.values(SEARCH_PAGE_CONFIG).find(
-        ({ type: configType }) => searchBarFilterValue === configType,
-      )
-      if (actionType) {
-        history.push({ pathname: actionType.path, search: query ? `?${query}` : '' })
-      }
-    }
 
-    // @ts-ignore
-    document.activeElement?.blur()
-  }
+      // @ts-ignore
+      document.activeElement?.blur()
+    },
+    [inputValue, selectedElement, searchBarFilterValue],
+  )
```

Comment:
I believe you can get rid of this `@ts-ignore`:

```suggestion
   if (document.activeElement instanceof HTMLElement) {
     document.activeElement.blur()
   }
```